temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -1060,13 +1060,15 @@ private void processFailWorkflowTask(
 
   // TODO: insert a single  workflow task timeout into the history
   private void timeoutWorkflowTask(long scheduledEventId) {
+    StickyExecutionAttributes previousStickySettings = this.stickyExecutionAttributes;
     try {
       completeWorkflowTaskUpdate(
           ctx -> {
             if (workflowTaskStateMachine == null
                 || workflowTaskStateMachine.getData().scheduledEventId != scheduledEventId
                 || workflowTaskStateMachine.getState() == State.NONE) {
               // timeout for a previous workflow task
+              stickyExecutionAttributes = previousStickySettings; // rollout sticky options
```

Comment:
Will this always go back to the non-sticky queue? That's what should happen in this situation, no?

Reply:
No, this branch checks if the timeout happened for a workflow task that is already done. We return sticky queue back in place.

Reply:
Ah, got it.